### ISSUE TITLE
Postpone checking a pattern let binding when type is blocked

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#3085]
Reported by Sergei Meshveliani on the Agda list:
```agda
open import Data.Product using (_×_; _,_)
open import Data.List    using (List; map)
open import Data.Nat     using (ℕ; suc)

f : List (ℕ × ℕ) → List (ℕ × ℕ)
f ps =
     map (\p → let (x , y) = p in (x , suc y))  ps
```
We should postpone checking a pattern let when the type is blocked on a meta, like we already do for pattern lambdas and top-level pattern matching definitions.

**Comment by asr** [comment]
CC'ing @mechvel.
